### ISSUE TITLE
 `:ownership_timeout` setting globally do not work

### ISSUE THREAD

**Issue by pthomalla** [elixir-ecto/ecto_sql#98]
### Environment
* Ecto version (mix deps): "3.0.7", "44dda84ac6b17bbbdeb8ac5dfef08b7da253b37a453c34ab1a98de7f7e5fec7f"
* Database adapter and version (mix deps):
ecto_sql, "3.0.5", "7e44172b4f7aca4469f38d7f6a3da394dbf43a1bcf0ca975e958cb957becd74e"
db_connection, "2.0.6", "bde2f85d047969c5b5800cb8f4b3ed6316c8cb11487afedac4aa5f93fd39abfa"
### Current behavior

https://github.com/elixir-ecto/ecto_sql/blob/b9dfe396e34beb6089e0cafdc2af2ee2b7a757ba/lib/ecto/adapters/sql/sandbox.ex#L212-L232

we drop `:ownership_timeout` option (ecto_sql/lib/ecto/adapters/sql.ex:450)

https://github.com/elixir-ecto/ecto_sql/blob/b9dfe396e34beb6089e0cafdc2af2ee2b7a757ba/lib/ecto/adapters/sql.ex#L427-L453

then when `Ecto.Adapters.SQL.checkout(DB.Repo)`  
`:ownership_timeout` options for my repo config in `config/test.exs` is not available. 

https://github.com/elixir-ecto/ecto_sql/blob/b9dfe396e34beb6089e0cafdc2af2ee2b7a757ba/lib/ecto/adapters/sql/sandbox.ex#L485-L486

### Expected behavior

don't clean up option in ecto_sql/lib/ecto/adapters/sql.ex

**Comment by josevalim** [comment]
Good find. Can you please send a PR that adds the ownership_timeout to the options? Thank you!

**Comment by artonragsdale** [comment]
I'm running `ecto_sql` 3.1.6, which I verified to have the fix in place from #62c9f68 but I still seem to be having issues with `:ownership_timeout` being set from config properly. Here is an error I got during a seeding operation, which shows the configuration settings having the config value 30000, while the error shows the default value of 60000:
```20:55:45.748 [error] Postgrex.Protocol (#PID<0.10358.0>) disconnected: ** (DBConnection.ConnectionError) owner #PID<0.91.0> timed out because it owned the connection for longer than 60000ms (set via the :ownership_timeout option)
** (exit) exited in: DBConnection.Holder.checkout(#PID<0.10403.0>, [log: #Function<11.128389407/1 in Ecto.Adapters.SQL.with_log/3>, source: "name_of_table", pool_size: 10, pool: DBConnection.Ownership, ownership_timeout: 300000, timeout: 300000])```
The stack trace follows:
```** (EXIT) no process: the process is not alive or there's no process currently associated with the given name, possibly because its application isn't started
    (db_connection) lib/db_connection/holder.ex:71: DBConnection.Holder.checkout/2
    (db_connection) lib/db_connection.ex:1030: DBConnection.checkout/3
    (db_connection) lib/db_connection.ex:1340: DBConnection.run/6
    (db_connection) lib/db_connection.ex:596: DBConnection.execute/4
    (ecto_sql) lib/ecto/adapters/postgres/connection.ex:80: Ecto.Adapters.Postgres.Connection.execute/4
    (ecto_sql) lib/ecto/adapters/sql.ex:572: Ecto.Adapters.SQL.execute!/4
    (ecto_sql) lib/ecto/adapters/sql.ex:554: Ecto.Adapters.SQL.execute/5
    (ecto) lib/ecto/repo/queryable.ex:153: Ecto.Repo.Queryable.execute/4```

**Comment by josevalim** [comment]
How are you setting it?
-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

**Comment by artonragsdale** [comment]
I'm setting it in `config/test.exs` like so:
```
config :app_name, AppName.Repo,
  adapter: Ecto.Adapters.Postgres,
  username: "postgres",
  password: "password",
  database: "app_name_test",
  hostname: "localhost",
  pool: Ecto.Adapters.SQL.Sandbox,
  ownership_timeout: 300_000,
  timeout: 300_000,
  log: false
```

**Comment by artonragsdale** [comment]
@josevalim I'm thinking it's likely because we're running the seeding operation when we reset the testing database using `MIX_ENV=test mix ecto.reset`. It appears that the `:ownership_timeout` setting is only applied during regular test checkout. Do you have any recommendation for doing something like that with a time consuming seeding baseline? Should we use a non-sandboxed database connection for the seeding somehow and then go back to the sandbox for the actual tests? I'm not sure how configuration should work for that.

**Comment by josevalim** [comment]
No idea from the top of my head. Can you provide a small app that reproduces it? That would speed up the process greatly. Thanks!

**Comment by artonragsdale** [comment]
@josevalim Thank you for taking the time to look into this. Here is an example project demonstrating the issue we're running into. :ownership_timeout does not appear to be applied when seeding the test database.

[https://github.com/artonragsdale/ecto-test-seed-timeout-example](https://github.com/artonragsdale/ecto-test-seed-timeout-example)

**Comment by artonragsdale** [comment]
I was thinking that it would make sense for the seeding operation to respect the :ownership_timeout settings since it is run by default when resetting the test database. Unless there is some way to use a non-sandboxed pool for the seeding.

**Comment by josevalim** [comment]
Thanks @artonragsdale , the app helped a lot. The issue has been fixed in DB Connection 2.1.1 which was just released.
